### PR TITLE
Cache Config

### PR DESCRIPTION
**Pull request by MGatner**
**Description**
Adds a new config value for reserved characters so developers may set their own restrictions.

**Checklist:**
- [X] Securely signed commits
- [X] Component(s) with PHPdocs
- [X] Unit testing, with >80% coverage
- [X] User guide updated
- [X] Conforms to style guide

### CODE REVIEW

**Comment by MGatner** [comment]
Effectively supersedes #4765

**Review by michalsn — APPROVED** [review]

File: app/Config/Cache.php [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 namespace Config;
 
+use CodeIgniter\Cache\Handlers\BaseHandler;
```
Comment (paulbalandan) [inline app/Config/Cache.php:5]: Why is this added? I don't remember it being used.

Comment (MGatner) [reply app/Config/Cache.php:5]: Oops! I was using it to reference the original value but then I used the string directly after I deprecated the constant. Good catch.

Comment (MGatner) [inline app/Config/Cache.php:5]:
```suggestion
```

**Review by paulbalandan — APPROVED** [review]
